libs/common/src/misc/serviceUtils.ts

```diff
@@ -53,15 +62,45 @@ export class ServiceUtils {
     }
   }
 
+  /**
+   * Searches a tree for a node with a matching `id`
+   * @param {TreeNode<ITreeNodeObject>} nodeTree - A single TreeNode branch that will be searched
+   * @param {string} id - The id of the node to be found
+   * @returns {TreeNode<ITreeNodeObject>} The node with a matching `id`
+   */
   static getTreeNodeObject(
+    nodeTree: TreeNode<ITreeNodeObject>,
+    id: string
+  ): TreeNode<ITreeNodeObject> {
+    if (nodeTree.node.id === id) {
+      return nodeTree;
+    }
+    for (let i = 0; i < nodeTree.children.length; i++) {
+      if (nodeTree.children[i].children != null) {
+        const node = ServiceUtils.getTreeNodeObject(nodeTree.children[i], id);
+        if (node !== null) {
+          return node;
+        }
+      }
+    }
+    return null;
+  }
+
+  /**
+   *
```

Comment:
Looks like you're missing a high-level description here.

```suggestion
   * Searches an array of tree nodes for a node with a matching `id`
```